packit_service/worker/parser.py

```diff
@@ -1424,13 +1424,17 @@ def parse_koji_task_event(event) -> Optional[koji.result.Task]:
             if children.get("method") == "buildArch":
                 rpm_build_task_ids[children.get("arch")] = children.get("id")
 
+        request = nested_get(event, "info", "request", default=[])
+        target_branch = request[1] if len(request) >= 2 else None
```

Comment:
I'm not having any luck finding documentation or source code for the `request` list, but inspecting a few messages it seems the second item is in fact a Koji build target name (e.g. `f41-candidate`, `f43`), not a branch name.

We do have [`KojiHelper.get_build_target()`](https://github.com/packit/packit/blob/9be956e56fccff773eaf1db0d60c12c739e211ed/packit/utils/koji_helper.py#L401) that converts branch name to build target so we could perhaps somehow use it to do the opposite and find a branch from a target, but I wonder if there is a more reliable way to get the matching branch.

Reply:
Oh right, I missed it 🙈 [here](https://koji.fedoraproject.org/koji/taskinfo?taskID=130730129) it is named *build target* (in the parameters list). So I think you are right. And I wasn't able to find anything else, in the event, to tell which was the target branch. If we can't reliably guess the branch from the target build (I don't know if we can?) then maybe we can save it in the database when submitting the build?

Reply:
I've opened https://github.com/packit/packit/pull/2571.

Reply:
> maybe we can save it in the database when submitting the build?

That's also an option. Not sure what's better.

Reply:
Ok, so we trigger the scratch builds with literal branch name as a build target, so in case of Packit-triggerd builds the build target in `request` should always be equal to a branch name. That means we don't have to do any conversion, however as the event object is generic, the attribute shouldn't be called `target_branch` but `build_target`.